packages/components/src/Card/Card.tsx

```diff
@@ -55,15 +85,45 @@ export function Card({
 
   const cardContent = (
     <>
-      {title && (
-        <div className={styles.header}>
-          <Heading level={3}>{title}</Heading>
-        </div>
-      )}
+      {CardHeader && CardHeader()}
       {children}
     </>
   );
 
+  function CardHeader() {
+    // Case 1: Deprecated Title
+    if (title) {
+      return (
+        <div className={styles.header}>
+          {title && <Heading level={3}>{title}</Heading>}
+        </div>
+      );
+    }
+    // Case 2: String Header
+    if (typeof header === "string") {
+      return (
+        <div className={styles.header}>
+          {header && <Heading level={3}>{header}</Heading>}
+        </div>
+      );
+    }
+    // Case 3: Custom Header
+    if (React.isValidElement(header)) {
+      return header;
+    }
+    // Case 4: Default Header Props
+    return (
+      !!header && (
```

Comment:
Is the `!!` required. I am just asking for my own learning

Reply:
it's not required here, I"m removing it to keep consistent with the rest